### PR TITLE
Docs: GC - cleanup

### PR DESCRIPTION
**Pull request by johnnyaug**
Split it into pages: committed, uncommited and "internals" which will include advanced topics.

Before:
image

After (committed page):
image

After (uncommitted):
image

### CODE REVIEW

File: docs/howto/garbage-collection.md [hunk 1]
```diff
@@ -1,38 +1,38 @@
 ---
```
Comment (itaiad200) [inline docs/howto/garbage-collection.md:1]: Maybe rename the file too?

File: docs/howto/garbage-collection-index.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
```
Comment (itaiad200) [inline docs/howto/garbage-collection-index.md:1]: Clicking on the index brings me to the `committed gc` page.

File: docs/howto/garbage-collection-index.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+layout: default
+title: Garbage Collection
+description: Clean up unnecessary objects using the garbage collection feature in lakeFS.
+parent: How-To
+nav_order: 20
+has_children: true
+redirect_from: ../reference/garbage-collection.html
+---
```
Comment (itaiad200) [inline docs/howto/garbage-collection-index.md:9]: Maybe we should give a summary and highlight the managed solution here?

**Review by itaiad200 — APPROVED** [review]
LGTM

File: docs/howto/garbage-collection-index.md (hunk 2 above)
Comment (johnnyaug) [reply docs/howto/garbage-collection-index.md:1]: It's a feature :)
Wanted to leave the default page as it is today, without an actual index page.

File: docs/howto/garbage-collection-index.md (hunk 3 above)
Comment (johnnyaug) [reply docs/howto/garbage-collection-index.md:9]: See other comment

File: docs/howto/garbage-collection.md (hunk 1 above)
Comment (johnnyaug) [reply docs/howto/garbage-collection.md]: Done

**Pull request merged**